ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time.c

```diff
@@ -132,10 +181,82 @@ static void sample(VALUE collector_instance) {
       (ddprof_ffi_Slice_label) {.ptr = NULL, .len = 0} // FIXME: TODO we need to gather the expected labels
     );
   }
+
+  // TODO: This seems somewhat overkill and inefficient to do every time we sample; to be improved later
+  remove_context_for_dead_threads(state);
```

Comment:
Will a simple counter do, given we know the sampling rate and thus can estimate a "don't do this more than once every X seconds"?

Reply:
I'll come back here with an actual decent response but for now I'll just drop the snarky version: [Hey look there's actually a new Ruby 3.2 feature for getting an event when threads start and end... Now I just need to wait for us to drop Ruby < 3.2 support :P](https://github.com/ruby/ruby/pull/6029)